src/classes/GRER_SyncGradeLevel_TDTM.cls

```diff
@@ -1,105 +1,199 @@
+        // A set of Contact Ids related to Grade Enrollments
         Set<Id> relatedContactIds = new Set<Id>();
-
-        //Declare and initiate set of current grade enrollment ids
+                                       
+        // A set of Current Grade Enrollment Ids
         Set<Id> currentGradeEnrolIds = new Set<Id>();
-        
-        //Declare and initiate lists to update
-        List<Contact> contactsToUpdate = new List<Contact>();
-        List<Grade_Enrollment__c> gradeEnrolsToUpdate = new List<Grade_Enrollment__c>();
-
-        //After INSERT and After UPDATE
-        if(triggerAction == hed.TDTM_Runnable.Action.AfterInsert 
-            || triggerAction == hed.TDTM_Runnable.Action.AfterUpdate) {
+                                       
+        //After INSERT
+        if (triggerAction == hed.TDTM_Runnable.Action.AfterInsert) {
+            if (listNew != null && listNew.size() > 0) {
+                List<Grade_Enrollment__c> newGradeEnrols = (List<Grade_Enrollment__c>)listNew;
+                selectGradeEnrolAndContactsToUpdate(newGradeEnrols, NULL, dmlWrapper);
+            }
+        }
+   
+        // After UPDATE
+        if (triggerAction == hed.TDTM_Runnable.Action.AfterUpdate) {
+        Map<ID, Grade_Enrollment__c> contactIdToOldGradeEnrolls = new Map<ID, Grade_Enrollment__c>();
+            if (listOld != NULL) {
+                contactIdToOldGradeEnrolls = new Map<ID, Grade_Enrollment__c>((List<Grade_Enrollment__c>)listOld);
+            }
             if (listNew != null && listNew.size() > 0) {
-                for (Integer i = 0; i < listNew.size(); i++) {
-                    Grade_Enrollment__c gradeErol = (Grade_Enrollment__c)listNew[i];
-                    if (gradeErol.Status__c != null 
-                        && gradeErol.Status__c == 'Current') {
-                        //Collect Ids of related contacts
-                        relatedContactIds.add(gradeErol.Contact__c);
-                        currentGradeEnrolIds.add(gradeErol.Id);
+                List<Grade_Enrollment__c> newGradeEnrols = (List<Grade_Enrollment__c>)listNew;
+                selectGradeEnrolAndContactsToUpdate(newGradeEnrols, contactIdToOldGradeEnrolls, dmlWrapper);
+            }
+        }
+                 
+        return dmlWrapper;
+    }
+    
+    private void selectGradeEnrolAndContactsToUpdate(List<Grade_Enrollment__c> newGradeEnrols, Map<ID, Grade_Enrollment__c> idToOldGradeEnrols, DmlWrapper dmlWrapper) {
+        Map<Id, Id> currentGradeEnrolToRelatedCon = new Map<Id, Id>();
+        
+        //A set of related Contact Ids that had its Grade Enrollments changed from a stauts of 'Current'
+        Set<Id> contactsIdsFormer = new Set<Id>();
+        
+        //A Map to capture Contacts and their related Grade Enrollments
+        Map<ID, List<Grade_Enrollment__c>> contactToGradeEnroll = new Map<ID, List<Grade_Enrollment__c>>();
+        
+        for (Integer i = 0; i < newGradeEnrols.size(); i++) {
```

Comment:
This actually was a previous code, not a part of this PR but since I moved after insert and after update to their own methods and added private methods it shows up here.